src/components/checkbox/checkbox.test.js

```diff
@@ -0,0 +1,557 @@
+      );
+      getComponent(testData).should("exist");
+    });
+
+    it("should render Checkbox component with data-element", () => {
+      CypressMountWithProviders(<CheckboxComponent data-element={testData} />);
+      getDataElementByValue(testData).should("exist");
+    });
+
+    it("should render Checkbox component with data-role", () => {
+      CypressMountWithProviders(<CheckboxComponent data-role={testData} />);
+      cy.get(`[data-role=${testData}]`).should("exist");
+    });
+
+    it.each([
+      [true, "be.checked"],
+      [false, "not.be.checked"],
+    ])(
+      "should render Checkbox component with checked state set to %s",
+      (booleanValue, assertion) => {
+        CypressMountWithProviders(<CheckboxComponent checked={booleanValue} />);
+        checkboxRole().should(assertion);
+      }
+    );
+  });
+
+  it.each(specialCharacters)(
+    "should render Checkbox component with %s as a label",
+    (checkboxLabel) => {
+      CypressMountWithProviders(<CheckboxComponent label={checkboxLabel} />);
+
+      label().should("have.text", checkboxLabel);
+    }
+  );
+
+  it.each(specialCharacters)(
+    "should render Checkbox component with %s as fieldHelp",
+    (fieldHelp) => {
+      CypressMountWithProviders(<CheckboxComponent fieldHelp={fieldHelp} />);
+
+      fieldHelpPreview().should("have.text", fieldHelp);
+    }
+  );
+
+  it("should render Checkbox component with inline fieldHelp", () => {
+    CypressMountWithProviders(
+      <CheckboxComponent fieldHelp="Inline fieldhelp" fieldHelpInline />
+    );
+
+    cy.get(
+      "[data-component=checkbox] > div:nth-child(1) > div > div > span"
+    ).should("have.text", "Inline fieldhelp");
+  });
+
+  it("should render Checkbox component with helpAriaLabel", () => {
+    CypressMountWithProviders(
+      <CheckboxComponent
+        label="Label For CheckBox"
+        labelHelp="Label Help"
+        helpAriaLabel="This text provides more information for the label"
+      />
+    );
+
+    icon().trigger("mouseover");
+    helpIcon()
+      .should("have.attr", "aria-label")
+      .and("contain", "This text provides more information for the label");
+  });
+
+  it.each([
+    [true, "be.disabled"],
+    [false, "not.be.disabled"],
+  ])(
+    "should render Checkbox component with disabled prop set to %s",
+    (booleanValue, assertion) => {
+      CypressMountWithProviders(<CheckboxComponent disabled={booleanValue} />);
+      checkboxRole().should(assertion);
+    }
+  );
+
+  it("should render Checkbox component with id", () => {
+    CypressMountWithProviders(
+      <CheckboxComponent id="f62d5e6e-248c-4b04-ae51-f7fab5bd93a3" />
+    );
+    checkboxRole().should("have.id", "f62d5e6e-248c-4b04-ae51-f7fab5bd93a3");
+  });
+
+  it("should render Checkbox component with name", () => {
+    CypressMountWithProviders(<CheckboxComponent name="confirm permission" />);
+    checkboxRole().should("have.attr", "name", "confirm permission");
```

Comment:
in the same way we should check `data-component`, `data-element` and `data-role`

Reply:
I have made similiar updates to data-component, data-element and data-role